pkg/app/piped/cloudprovider/lambda/client.go

```diff
@@ -408,6 +412,137 @@ func (c *client) UpdateTrafficConfig(ctx context.Context, fm FunctionManifest, r
 	return nil
 }
 
+func (c *client) updateTagsConfig(ctx context.Context, fm FunctionManifest) error {
+	getFuncInput := &lambda.GetFunctionInput{
+		FunctionName: aws.String(fm.Spec.Name),
+	}
+	output, err := c.client.GetFunctionWithContext(ctx, getFuncInput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case lambda.ErrCodeInvalidParameterValueException:
+				return fmt.Errorf("invalid parameter given: %w", err)
+			case lambda.ErrCodeServiceException:
+				return fmt.Errorf("aws lambda service encountered an internal error: %w", err)
+			case lambda.ErrCodeTooManyRequestsException:
+				return fmt.Errorf("request throughput limit was exceeded: %w", err)
+			case lambda.ErrCodeResourceNotFoundException:
+				return fmt.Errorf("resource not found: %w", err)
+			}
+		}
+		return fmt.Errorf("unknown error occurred on list tags of Lambda function %s: %w", fm.Spec.Name, err)
+	}
+
+	functionArn := aws.StringValue(output.Configuration.FunctionArn)
+	currentTags := aws.StringValueMap(output.Tags)
+	// Skip if there are no changes on tags.
+	if reflect.DeepEqual(currentTags, fm.Spec.Tags) {
+		return nil
+	}
+
+	newDefinedTags := make(map[string]string)
+	updatedTags := make(map[string]string)
+	removedTags := make(map[string]string)
+	for k, v := range fm.Spec.Tags {
+		val, ok := currentTags[k]
+		if !ok {
+			newDefinedTags[k] = v
+		}
+		if val != v {
+			updatedTags[k] = v
+		}
+	}
+	for k, v := range currentTags {
+		_, ok := fm.Spec.Tags[k]
+		if !ok {
+			removedTags[k] = v
+		}
+	}
```

Comment:
Can you move this part into a new function and add some tests for it?

Reply:
thank you 🙏 
Addressed on https://github.com/pipe-cd/pipe/pull/1596/commits/7b0196c568bf7853807d92ae5a69e8d57e64a9ee 😁